src/askvm/resources/core/forIn.ts

```diff
@@ -1,11 +1,52 @@
 import { any, resource, run, runUntyped } from '../../lib';
+import { isAskCode } from '../../../askcode';
 
 export const forIn = resource({
   type: any,
   async compute(options, code, args) {
     const { params } = code;
-    const [condition, args2, block] = params!;
-    const value = await runUntyped(options, condition);
-    return run(options, value ? block : null, []);
+    if (!params || params?.length !== 3) {
+      throw new Error('forIn expects exactly 3 params');
+    }
+    const [key, of, block] = params;
+    if (key === null || of === null || block === null) {
+      throw new Error('forIn expects non-null key, in and block');
+    }
+    if (!isAskCode(key)) {
+      throw new Error('forIn expects key to be a statement');
+    }
+    if (!['let', 'const'].includes(key.name)) {
+      throw new Error('forIn expects key to be a "let" or "const" statement');
+    }
+    if (!Array.isArray(key.params)) {
+      throw new Error('forIn expects key to have params!');
+    }
```

Comment:
Nice error handling 👍

Reply:
Thanks :) It would be nice to have good error handling in the whole Ask.